src/common/snippets/src/lowered/pass/init_loops.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/init_loops.hpp"
+
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/itt.hpp"
+
+namespace ngraph {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+namespace {
+void filter_ports(LinearIR& linear_ir,
+                  std::vector<ExpressionPort>& loop_entries, std::vector<ExpressionPort>& loop_exits) {
+    std::vector<ExpressionPort> new_loop_entries;
+    std::vector<ExpressionPort> new_loop_exits;
+    new_loop_entries.reserve(loop_entries.size());
+    new_loop_exits.reserve(loop_exits.size());
+
+    std::set<std::shared_ptr<ov::Node>> loop_parents;
+    for (const auto& loop_entry_point : loop_entries) {
+        const auto& expr = loop_entry_point.expr;
+        const auto port = loop_entry_point.port;
+        const auto node = expr->get_node();
+        const auto ma = ov::as_type_ptr<op::MemoryAccess>(node);
+        if (ma && ma->is_memory_access_input_port(port)) {
+            const auto& parent_expr = linear_ir.get_expr_by_output(expr->get_inputs()[port]).expr;
+            const auto& parent = parent_expr->get_node();
+            // Todo: Sometimes several Load in one Loop read data from the same Node
```

Comment:
It's about cases when several `Load` expr can read data from the same source. The same situation like with `Buffer` -> we have double ptr increments and finalization offsets. Here we don't take the `Load` if there is already another `Load` expression wit the same parent.
But honestly, it's incorrect way. We should cover these cases as for `Buffer`. So I updated the previous pass `ResetBuffers` to add support for `Load` (it's not needed to `Store` because we cannot write to the same data several times from different `Store`).